diff -ur css/template.css css/template.css
--- css/template.css	2012-08-13 11:56:01.000000000 +0100
+++ css/template.css	2015-01-04 18:34:30.000000000 +0000
@@ -503,3 +503,14 @@
         width: 203px;
     }
 }
+
+/* Needed for Composr's large number of packages */
+
+.dropdown-menu {
+	min-width: 420px;
+}
+
+.dropdown-menu li {
+	display: inline-block;
+	min-width: 130px;
+}
diff -ur index.xsl index.xsl
--- index.xsl	2012-08-13 15:41:22.000000000 +0100
+++ index.xsl	2015-01-04 18:53:02.000000000 +0000
@@ -14,18 +14,7 @@
 
         <div class="row">
             <div class="span7">
-                <xsl:if test="count(/project/namespace[@name != 'default']) > 0">
-                <div class="well">
-                    <ul class="nav nav-list">
-                        <li class="nav-header">Namespaces</li>
-                        <xsl:apply-templates select="/project/namespace" mode="menu">
-                            <xsl:sort select="@full_name" />
-                        </xsl:apply-templates>
-                    </ul>
-                </div>
-                </xsl:if>
-
-                <xsl:if test="count(/project/package[@name != '' and @name != 'default']) > 0">
+                 <xsl:if test="count(/project/package[@name != '' and @name != 'default']) > 0">
                 <div class="well">
                     <ul class="nav nav-list">
                         <li class="nav-header">Packages</li>
@@ -38,21 +27,22 @@
 
             </div>
             <div class="span5">
-					<b>Welcome to the ocPortal API documentation.</b><br />
+					<b>Welcome to the Composr API documentation.</b><br />
 					<br />
-					This documentation is consistent with version 5.1 of ocPortal. For other documentation (including our developers guide, the Code Book), see the <a target="_blank" title="(this link will open in a new window)" href="http://ocportal.com/docs/">documentation section of our website</a>.
+					For other documentation (including our developers guide, the Code Book), see the <a target="_blank" title="Documentation section of our website (this link will open in a new window)" href="https://compo.sr/docs/">documentation section of our website</a>.
 					<br />
 					<br />
-					The packages in this documentation (shown above) correspond to the addons that ocPortal is split up into. All packages beginning 'core_' can be assumed to be installed on any ocPortal version.
+					The packages in this documentation (shown above) correspond to the addons that Composr is split up into. All packages beginning 'core_' can be assumed to be installed on any Composr version.
 					<br />
 					<br />
-					Be aware that you still often need to use the 'require_code' command to gain access to source code files (e.g. access <tt>sources/files.php</tt> using<br /><code>require_code('files');</code>.<br />Some source files are always loaded up, see the 'init' function of <tt>sources/global2.php</tt> to discover which.<br />
-					Hooks are called up by whatever code using the hooks.<br />
-					Modules are called up on-demand by users as pages (but aren't included in this API documentation).<br />
+					Be aware that you still often need to use the 'require_code' command to gain access to source code files (e.g. access <tt>sources/files.php</tt> using<br /><code>require_code('files');</code>.<br /><br />Some source files are always loaded up, see the 'init' function of <tt>sources/global2.php</tt> to discover which.<br /><br />
+					Hooks are called up by whatever code using the hooks.<br /><br />
+					Modules are called up on-demand by users as pages (but aren't included in this API documentation).<br /><br />
 					Blocks are called up by placement on pages using Comcode, or in templates using Tempcode.
 					<br />
 					<br />
-					To find the forum driver API you can look at <em>core_forum_drivers</em>, but also look at the base <strong>forum_driver_base</strong> class under <em>core</em>.
+					<br />
+					To find the forum driver API you can look at <em>core_forum_drivers</em>, but also look at the base <strong>Forum_driver_base</strong> class under <em>core</em>.
 					<br />
 					To find the database API look at the <strong>database_driver</strong> class under <em>core</em>.
             </div>
diff -ur js/template.js js/template.js
--- js/template.js	2012-08-13 11:56:01.000000000 +0100
+++ js/template.js	2015-01-04 18:34:30.000000000 +0000
@@ -90,14 +90,14 @@
     // chrome cannot deal with certain situations; warn the user about reduced features
     if ($.browser.chrome && (window.location.protocol == 'file:')) {
         $("body > .container").prepend(
-            '<div class="alert alert-error"><a class="close" data-dismiss="alert">×</a>' +
+            '<div class="alert alert-error"><a class="close" data-dismiss="alert">&cross;</a>' +
             'You are using Google Chrome in a local environment; AJAX interaction has been ' +
             'disabled because Chrome cannot <a href="http://code.google.com/p/chromium/issues/detail?id=40787">' +
             'retrieve files using Ajax</a>.</div>'
         );
     }
 
-    $('ul.nav-namespaces li a, ul.nav-packages li a').click(function(){
+    /*Makes navigation in Composr nasty$('ul.nav-namespaces li a, ul.nav-packages li a').click(function(){
         // Google Chrome does not do Ajax locally
         if ($.browser.chrome && (window.location.protocol == 'file:'))
         {
@@ -120,7 +120,7 @@
         );
 
         return false;
-    });
+    });*/
 
     function filterPath(string)
     {
diff -ur layout.xsl layout.xsl
--- layout.xsl	2012-08-13 15:40:40.000000000 +0100
+++ layout.xsl	2015-01-04 18:34:30.000000000 +0000
@@ -82,8 +82,7 @@
                         <span class="icon-bar"></span> <span class="icon-bar"></span>
                         <span class="icon-bar"></span> </a>
                     <a class="brand" href="{$root}index.html">
-                        <xsl:value-of select="$title" disable-output-escaping="yes"/>
-                        <xsl:if test="$title = ''">phpDocumentor</xsl:if>
+                        Composr API documentation
                     </a>
 
                     <div class="nav-collapse">
@@ -137,9 +136,7 @@
         <meta name="viewport" content="width=device-width; initial-scale=1.0; maximum-scale=1.0; user-scalable=0;" />
         <meta charset="utf-8" />
         <title>
-            <xsl:value-of select="$title" disable-output-escaping="yes" />
-            <xsl:if test="$title = ''">phpDocumentor</xsl:if>
-            <xsl:apply-templates select="." mode="title"/>
+            Composr API documentation
         </title>
         <meta name="author" content="Mike van Riel" />
         <meta name="description" content="" />
diff -ur template.xml template.xml
--- template.xml	2012-08-13 11:56:01.000000000 +0100
+++ template.xml	2015-01-04 19:39:36.000000000 +0000
@@ -20,18 +20,18 @@
   ]]></description>
   <transformations>
     <transformation query="copy" writer="FileIo" source="images" artifact="img"/>
-    <transformation query="copy" writer="FileIo" source="templates/responsive/img" artifact="img"/>
-    <transformation query="copy" writer="FileIo" source="templates/responsive/css" artifact="css"/>
+    <transformation query="copy" writer="FileIo" source="templates/composr-api-template/img" artifact="img"/>
+    <transformation query="copy" writer="FileIo" source="templates/composr-api-template/css" artifact="css"/>
     <transformation query="copy" writer="FileIo" source="js" artifact="js"/>
-    <transformation query="copy" writer="FileIo" source="templates/responsive/js" artifact="js"/>
-    <transformation query="" writer="xsl" source="templates/responsive/index.xsl" artifact="index.html"/>
-    <transformation query="//namespace/@full_name" writer="xsl" source="templates/responsive/namespaces.xsl" artifact="namespaces/{$full_name}"/>
-    <transformation query="//package/@full_name" writer="xsl" source="templates/responsive/packages.xsl" artifact="packages/{$full_name}"/>
-    <transformation query="//class/full_name|//interface/full_name" writer="xsl" source="templates/responsive/class.xsl" artifact="classes/{$full_name}"/>
-    <transformation query="" writer="xsl" source="templates/responsive/reports/markers.xsl" artifact="markers.html"/>
-    <transformation query="" writer="xsl" source="templates/responsive/reports/errors.xsl" artifact="errors.html"/>
-    <transformation query="" writer="xsl" source="templates/responsive/reports/deprecated.xsl" artifact="deprecated.html"/>
-    <transformation query="" writer="xsl" source="templates/responsive/graphs/class.xsl" artifact="graph_class.html"/>
+    <transformation query="copy" writer="FileIo" source="templates/composr-api-template/js" artifact="js"/>
+    <transformation query="" writer="xsl" source="templates/composr-api-template/index.xsl" artifact="index.html"/>
+    <transformation query="//namespace/@full_name" writer="xsl" source="templates/composr-api-template/namespaces.xsl" artifact="namespaces/{$full_name}"/>
+    <transformation query="//package/@full_name" writer="xsl" source="templates/composr-api-template/packages.xsl" artifact="packages/{$full_name}"/>
+    <transformation query="//class/full_name|//interface/full_name" writer="xsl" source="templates/composr-api-template/class.xsl" artifact="classes/{$full_name}"/>
+    <transformation query="" writer="xsl" source="templates/composr-api-template/reports/markers.xsl" artifact="markers.html"/>
+    <transformation query="" writer="xsl" source="templates/composr-api-template/reports/errors.xsl" artifact="errors.html"/>
+    <transformation query="" writer="xsl" source="templates/composr-api-template/reports/deprecated.xsl" artifact="deprecated.html"/>
+    <transformation query="" writer="xsl" source="templates/composr-api-template/graphs/class.xsl" artifact="graph_class.html"/>
     <transformation query="" writer="Graph" source="Class" artifact="classes.svg" />
   </transformations>
 </template>
