diff --git a/billing_inc.php b/billing_inc.php
index 0356a77..d5f13ef 100644
--- a/billing_inc.php
+++ b/billing_inc.php
@@ -75,7 +75,7 @@
 	# CSRF protection not required here - form does not result in modifications
 ?>
 
-<form method="post" action="<?php echo form_action_self() ?>">
+<form method="post" action="<?php echo string_attribute( form_action_self() ) ?>">
 	<input type="hidden" name="id" value="<?php echo isset( $f_bug_id ) ? $f_bug_id : 0 ?>" />
 	<table border="0" class="width100" cellspacing="0">
 		<tr>
diff --git a/bugnote_stats_inc.php b/bugnote_stats_inc.php
index 3c1d4c9..4477d58 100644
--- a/bugnote_stats_inc.php
+++ b/bugnote_stats_inc.php
@@ -65,7 +65,7 @@
 	# CSRF protection not required here - form does not result in modifications
 ?>
 
-<form method="post" action="<?php echo form_action_self() . '#bugnotestats' ?>">
+<form method="post" action="<?php echo string_attribute( form_action_self() . '#bugnotestats' ) ?>">
 	<input type="hidden" name="id" value="<?php echo $f_bug_id ?>" />
 	<table border=0 class="width100" cellspacing="0">
 		<tr>
diff --git a/core/authentication_api.php b/core/authentication_api.php
index 94d37be..7b363a4 100644
--- a/core/authentication_api.php
+++ b/core/authentication_api.php
@@ -694,7 +694,7 @@ function auth_reauthenticate_page( $p_user_id, $p_username ) {
 	}
 	?>
 </p>
-<form name="reauth_form" method="post" action="<?php echo form_action_self();?>">
+<form name="reauth_form" method="post" action="<?php echo string_attribute( form_action_self() ) ?>">
 <?php
 	# CSRF protection not required here - user needs to enter password
 	# (confirmation step) before the form is accepted.
diff --git a/core/form_api.php b/core/form_api.php
index c81780e..4467133 100644
--- a/core/form_api.php
+++ b/core/form_api.php
@@ -35,7 +35,8 @@
  * @return string Form action value
  */
 function form_action_self() {
-	return basename($_SERVER['PHP_SELF']);
+	$t_self = trim( str_replace( "\0", '', $_SERVER['SCRIPT_NAME'] ) );
+	return basename( $t_self );
 }
 
 /**
diff --git a/core/helper_api.php b/core/helper_api.php
index fd54372..09ef56d 100644
--- a/core/helper_api.php
+++ b/core/helper_api.php
@@ -299,7 +299,7 @@ function helper_ensure_confirmed( $p_message, $p_button_label ) {
 	print_hr();
 	echo "\n$p_message\n";
 
-	echo '<form method="post" action="' . form_action_self() . "\">\n";
+	echo '<form method="post" action="' . string_attribute( form_action_self() ) . "\">\n";
 	# CSRF protection not required here - user needs to confirm action
 	# before the form is accepted.
 	print_hidden_inputs( gpc_strip_slashes( $_POST ) );
diff --git a/manage_config_email_page.php b/manage_config_email_page.php
index 34d9ef3..fe04ff3 100644
--- a/manage_config_email_page.php
+++ b/manage_config_email_page.php
@@ -286,7 +286,7 @@ function get_section_end_for_email() {
 			echo form_security_field( 'manage_config_revert' );
 			echo "<input name=\"revert\" type=\"hidden\" value=\"notify_flags,default_notify_flags\"></input>";
 			echo "<input name=\"project\" type=\"hidden\" value=\"$t_project\"></input>";
-			echo "<input name=\"return\" type=\"hidden\" value=\"" . form_action_self() ."\"></input>";
+			echo "<input name=\"return\" type=\"hidden\" value=\"" . string_attribute( form_action_self() ) ."\"></input>";
 			echo "<input type=\"submit\" class=\"button\" value=\"";
 			if ( ALL_PROJECTS == $t_project ) {
                 echo lang_get( 'revert_to_system' );
diff --git a/manage_config_work_threshold_page.php b/manage_config_work_threshold_page.php
index 6ec7747..5cd17ce 100644
--- a/manage_config_work_threshold_page.php
+++ b/manage_config_work_threshold_page.php
@@ -326,7 +326,7 @@ function get_section_end() {
 		echo form_security_field( 'manage_config_revert' );
         echo "<input name=\"revert\" type=\"hidden\" value=\"" . implode( ',', $t_overrides ) . "\"></input>";
         echo "<input name=\"project\" type=\"hidden\" value=\"$t_project_id\"></input>";
-        echo "<input name=\"return\" type=\"hidden\" value=\"" . form_action_self() ."\"></input>";
+        echo "<input name=\"return\" type=\"hidden\" value=\"" . string_attribute( form_action_self() ) ."\"></input>";
         echo "<input type=\"submit\" class=\"button\" value=\"";
         if ( ALL_PROJECTS == $t_project_id ) {
             echo lang_get( 'revert_to_system' );
diff --git a/manage_config_workflow_page.php b/manage_config_workflow_page.php
index eba3676..f316716 100644
--- a/manage_config_workflow_page.php
+++ b/manage_config_workflow_page.php
@@ -477,7 +477,7 @@ function access_row() {
 			echo form_security_field( 'manage_config_revert' );
             echo "<input name=\"revert\" type=\"hidden\" value=\"" . implode( ',', $t_overrides ) . "\"></input>";
             echo "<input name=\"project\" type=\"hidden\" value=\"$t_project\"></input>";
-            echo "<input name=\"return\" type=\"hidden\" value=\"" . form_action_self() ."\"></input>";
+            echo "<input name=\"return\" type=\"hidden\" value=\"" . string_attribute( form_action_self() ) ."\"></input>";
             echo "<input type=\"submit\" class=\"button\" value=\"";
             if ( ALL_PROJECTS == $t_project ) {
                 echo lang_get( 'revert_to_system' );
